### PR TITLE
Sort crate features by name

### PR DESCRIPTION
**Pull request by apiraino**
This patch changes one detail implemented in #1144. The feature flags
sorting was by "number of subfeatures". Now it is by feature alphabetically
which I find suitable to easily discover the feature I'm looking for in a crate.
It also ensures that the special "default" feature stays on top of the list.

If we don't want to touch this behaviour, I'm also 100% ok to decline.
If there is a better way to implement this I'm happy to improve it.

cc @almusil since author of the original patch
super-thanks to @GuillaumeGomez for mentoring me on this!

r? @GuillaumeGomez or @jyn514

### CODE REVIEW

File: src/web/features.rs [hunk 1]
```diff
@@ -79,15 +80,22 @@ pub fn build_features_handler(req: &mut Request) -> IronResult<Response> {
...
 
     features.extend(remaining.into_iter().rev());
+    features.sort_by(|a, b| {
+        // 'default' feature is special and should always stay on top
+        if b.name == "default" {
+            std::cmp::Ordering::Greater
+        } else {
+            a.name.partial_cmp(&b.name).unwrap()
+        }
+    });
```
Comment (apiraino) [inline src/web/features.rs:98]: at first I've tried to simply change the sorting at the SQL level on [line 55](https://github.com/rust-lang/docs.rs/pull/1320/files#diff-0c19324803df42be471baabdc27c2f8f371245f7626c7534bfb17de43872dc5eR55) but I think this sorting in this function would neutralize it anyway

**Comment by apiraino** [comment]
How the sorting looks before this patch

<details>
old
</details>

How the sorting looks after this patch
<details>
new
</details>

**Review by GuillaumeGomez — APPROVED** [review]
Nice work!

**Comment by almusil** [comment]
Thank you for including me.

I don't mind this change, I am not sure if the requirement has changed since the initial implementation.
However if we go with this, there is actually no need to compute the whole tree structure etc. 
This is maintainers call but I don't see a point in keeping code that is not used.

**Comment by apiraino** [comment]
> I don't mind this change, I am not sure if the requirement has changed since the initial implementation.

@almusil good point, I forgot to give some context. Example: I went to the tokio crate feature list (see my screenshots) looking for a specific one and I was confused because I expected them in alphabetical order, so I had to run `CTRL+F`and find it.

This may be a subjective "feeling" on how I see the sorting and this PR is just a small question+proposal, this is why I mentioned it's perfectly fine to decline this PR :)

Meta-thought: it's often hard to please every and each user, often there are conflicting thoughts about a UX feature :) The expectations of a single user (in this case: me) should not put anyone under pressure to accept a PR.

**Comment by almusil** [comment]
> > I don't mind this change, I am not sure if the requirement has changed since the initial implementation.
> 
> @almusil good point, I forgot to give some context. Example: I went to the tokio crate feature list (see my screenshots) looking for a specific one and I was confused because I expected them in alphabetical order, so I had to run `CTRL+F`and find it.

The idea was to have flat tree structure of features that are enabled by default followed by features that enable any other feature. 
But having them sorted by name also makes sense in wider context. 

> 
> This may be a subjective "feeling" on how I see the sorting and this PR is just a small question+proposal, this is why I mentioned it's perfectly fine to decline this PR :)

I understand that, this is worth a discussion.

> 
> Meta-thought: it's often hard to please every and each user, often there are conflicting thoughts about a UX/UI :) The expectations of a single user (in this case: me) should not put anyone under pressure to accept a PR.

**Comment by jyn514** [comment]
Personally I like the tree structure better, I think it shows the features you'd want most often towards the top. If the crate has really complicated features, I think it should be documenting those itself anyway.

**Comment by apiraino** [comment]
> Personally I like the tree structure better, I think it shows the features you'd want most often towards the top. If the crate has really complicated features, I think it should be documenting those itself anyway.

Based on your reasoning, I think you have a stronger case to keep it as it is, so I'll close my proposal.

(thanks @jyn514 for your thoughts on the matter)

**Pull request closed without merging**